### PR TITLE
feijoa:  l1infotree recursive

### LINKED ISSUE #3465: feijoa: add support for recursive L1InfoTree
# Rationale
New Feijoa fork implements Recursive l1InfoTree

### PR DESCRIPTION
**Pull request by joanestebanr**
Closes #3465

### Reviewers

Main reviewers:

- @John
- @Doe

Codeowner reviewers:

- @-Alice
- @-Bob

### CODE REVIEW

File: synchronizer/actions/etrog/processor_l1_info_tree_update.go [hunk 1]
```diff
@@ -24,9 +24,9 @@ type ProcessorL1InfoTreeUpdate struct {
 // NewProcessorL1InfoTreeUpdate new processor for GlobalExitRootsOrder
 func NewProcessorL1InfoTreeUpdate(state stateProcessorL1InfoTreeInterface) *ProcessorL1InfoTreeUpdate {
 	return &ProcessorL1InfoTreeUpdate{
-		ProcessorBase: actions.ProcessorBase[ProcessorL1InfoTreeUpdate]{
-			SupportedEvent:    []etherman.EventOrder{etherman.L1InfoTreeOrder},
-			SupportedForkdIds: &actions.ForksIdAll},
+		ProcessorBase: *actions.NewProcessorBase[ProcessorL1InfoTreeUpdate](
```
Comment (joanestebanr) [inline synchronizer/actions/etrog/processor_l1_info_tree_update.go:27]: What is the reason for this change?

Comment (tclemos) [reply synchronizer/actions/etrog/processor_l1_info_tree_update.go:27]: Just a refactoring to match the way I understood the `ProcessorBase` struct was intended to be implemented.
There were some methods to expose the fields adding extra logic when accessing these fields, but the fields were public and, there was no constructor to the struct, it was mixed publicly and privately implemented, then I adjusted it to work in a private way. 

This was done in the middle of other changes to the code, we can review it and change it if necessary.

**Review by tclemos — APPROVED** [review]

File: state/l1infotree_test.go [hunk 2]
```diff
@@ -45,7 +45,11 @@ func TestFirstLeafOfL1InfoTreeIsIndex0(t *testing.T) {
 	if err != nil {
 		panic(err)
 	}
-	testState := state.NewState(stateCfg, storage, nil, nil, nil, mt)
+	mtr, err := l1infotree.NewL1InfoTreeRecursive(32)
+	if err != nil {
+		panic(err)
```
Comment (ARR552) [inline state/l1infotree_test.go:50]:
```suggestion
		require.NoError(t, err)
```
